### PR TITLE
Add option to nyaml2nxdl to add space to self-closing tags

### PR DESCRIPTION
**Pull request by PeterC-DLS**
Also add short options

### CODE REVIEW

File: pynxtools/nyaml2nxdl/nyaml2nxdl.py [hunk 1]
```diff
@@ -148,33 +148,44 @@ def split_name_and_extension(file_name):
...
 possible issues in yaml files'
 )
-def launch_tool(input_file, verbose, append, check_consistency):
+@click.option(
+    '-s',
+    '--self-closing-tag-space',
+    is_flag=True,
+    default=False,
+    help='Ensure there is a single space character before all self-closing XML tag'
+)
```
Comment (sherjeelshabih) [inline pynxtools/nyaml2nxdl/nyaml2nxdl.py:187]: Do we need this as a flag? Wouldn't it be better to have this as the default?

**Review by RubelMozumder — COMMENTED** [review]
!!

File: pynxtools/nyaml2nxdl/nyaml2nxdl.py (hunk 1 above)
Comment (PeterC-DLS) [reply pynxtools/nyaml2nxdl/nyaml2nxdl.py:187]: I will change and flip the logic.

**Comment by sherjeelshabih** [comment]
Thank your for the changes, @PeterC-DLS. Flipping the logic looks good too. I'll leave the final review to @RubelMozumder and @sanbrock before the merge.

I was thinking that we could just not have such a flag and just add the space in self-closing tags as that seems to be the norm usually. But it really is a choice so I leave it up to the three of you to decide what is the best way to proceed.

**Comment by RubelMozumder** [comment]
It is a matter of choice to have space before the self-closing tag. Now, I assume that with the '-s' option no space before the closing tag, otherwise a default. So, the choice is up to the users on how they want to see XML files.
I would say this makes sense, right? 
So, my flag is ready to go to the master branch. Thanks to @sherjeelshabih for keeping me in conversation.

**Comment by sanbrock** [comment]
the code has already been migrated to a for of nexus_definitions ([FAIRmat-NFDI:nyaml2nxdl-migration](https://github.com/FAIRmat-NFDI/nexus_definitions/tree/nyaml2nxdl-migration)). Can you please migrate this PR there?

**Pull request closed without merging**